### ISSUE TITLE
2020/3/4-5

### ISSUE THREAD

**Issue by mingxoxo** [mingxoxo/deapLearning-study#6]
Optimizer를 정의하려면 초기화해야하는 변수와 learning rate를 알아야 한다.
![image](https://user-images.githubusercontent.com/44529556/76001406-667d6780-5f48-11ea-940c-337a70708ddd.png)

optimizer.zero_grad() : optimizer에 저장되어 있는 모든 학습가능한 변수들의 gradient를 0으로 초기화시킨다.
cost.backward() : cost function을 미분해서 각 변수들의 gradient 변수들을 채운다.
optimizer.step() : 저장된 gradient로 gradient descent를 실행

**Full code**
```
import torch
import torch.optim as optim
#데이터
x_train = torch.FloatTensor([[1], [2], [3]])
y_train = torch.FloatTensor([[1], [2], [3]])
#모델 초기화
W = torch.zeros(1, requires_grad=True)
#optimizer 설정
optimizer = optim.SGD([W], lr = 0.15)

#Epoch : 데이터로 학습한 횟수
nb_epochs = 10

for epoch in range(nb_epochs + 1) :

  #H(x) 계산
  hypothesis = x_train * W

  #cost 계산
  cost = torch.mean((hypothesis - y_train) ** 2)
  gradient = torch.sum((W * x_train - y_train)*x_train)

  print('Epoch {:4d}/{} W: {:.3f}, Cost: {:.6f}'.format(epoch, nb_epochs, W.item(), cost.item()))

  #cost로 H(x) 개선
  optimizer.zero_grad()
  cost.backward()
  optimizer.step()
  ```

**Comment by mingxoxo** [comment]
- [ ] optimizer.zero_grad()를 학습가능한 변수를 0으로 초기화하는 것인데 왜 for문에서 계속 초기화를 시키는 것인가 ?
-> 이게 아니라 gradient를 초기화 시키는거였다 바보다..
